geopandas/geoseries.py

```diff
@@ -74,6 +68,17 @@ def _geoseries_expanddim(data=None, *args, **kwargs):
     return df
 
 
+def _geoseries_expanddim(data=None, *args, **kwargs):
+    # pd.Series._constructor_expanddim == pd.DataFrame
+    df = pd.DataFrame(data, *args, **kwargs)
+    if isinstance(data, GeoSeries):
+        # pandas default column name is 0, keep convention
+        geo_col_name = data.name if data.name is not None else 0
```

Comment:
Can we add a test covering this line?

Reply:
This also doesn't seem to be covered on main. This block was added in https://github.com/geopandas/geopandas/pull/2296, which was a complex PR and includes a bunch of tests (including a test that ensures `to_frame()` returns a GeoDataFrame with "0" as column name), but so maybe this wasn't needed in the final state of that PR (it went through some iterations)

Reply:
Update: appears to be redundant in my local tests. Seems plausible to me it could have been redundant in that PR in the end.

Had a quick look and it appears we still have codecov corresponding to that branch, which indicates it was covered then (https://app.codecov.io/gh/geopandas/geopandas/commit/aa80aa851ad5ee385aa67e7c153c9f3626b11951/blob/geopandas/geoseries.py).

I'm guessing this was then some compatibility thing for an old version of pandas which we no longer test against.